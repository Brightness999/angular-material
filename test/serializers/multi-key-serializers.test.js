/**
 * @fileoverview Test multi-key serializers.
 */
const Logality = require('../../app/logality');
const { sink, stubLogality } = require('../lib/tester.lib');

const DATA = {
  id: 12,
  property: 'zero',
  value: 0.23,
};

describe('Multi-Key Serializers', () => {
  stubLogality();

  test('Will log multi-key serializer on context', done => {
    const serializers = {
      zit: data => {
        return [
          {
            path: 'context.zit',
            value: data,
          },
          {
            path: 'context.nit',
            value: data,
          },
          {
            path: 'rootpath',
            value: data,
          },
        ];
      },
    };

    const logality = new Logality({
      appName: 'testLogality',
      serializers,
      wstream: sink(chunk => {
        expect(chunk).toMatchSnapshot();
        done();
      }),
    });

    const log = logality.get();

    log('info', 'hello world', { zit: DATA });
  });
});
