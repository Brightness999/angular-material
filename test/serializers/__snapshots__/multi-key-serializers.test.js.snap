// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Multi-Key Serializers Will log multi-key serializer on context 1`] = `
Object {
  "context": Object {
    "nit": Object {
      "id": 12,
      "property": "zero",
      "value": 0.23,
    },
    "runtime": Object {
      "application": "testLogality",
    },
    "source": Object {
      "file_name": "/test/serializers/multi-key-serializers.test.js",
    },
    "system": Object {
      "hostname": "localhost",
      "pid": 36255,
      "process_name": "node .",
    },
    "zit": Object {
      "id": 12,
      "property": "zero",
      "value": 0.23,
    },
  },
  "dt": "2018-05-18T16:25:57.815Z",
  "event": Object {},
  "level": "info",
  "message": "hello world",
  "rootpath": Object {
    "id": 12,
    "property": "zero",
    "value": 0.23,
  },
  "severity": 6,
}
`;
